### PR TITLE
[IMP] doc: improves commit guidelines

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
This commit:
- removes deprecated guidelines for static file and
assets struture
- fix some typo and mistakes

### CODE REVIEW

File: doc/reference/guidelines.rst [hunk 1]
```diff
@@ -963,17 +961,9 @@ Javascript coding guidelines
...
-  element which is specific to the pages your code needs to run on
-  using JQuery.
-
-.. code-block:: javascript
-
-    odoo.website.if_dom_contains('.jquery_class_selector', function () {
-        /*your code here*/
-    });
 
+Everything you need to know is explain at https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines. You can also need
```
Comment (pedrobaeza) [inline doc/reference/guidelines.rst]: s/explain/explained

File: doc/reference/guidelines.rst [hunk 2]
```diff
@@ -71,8 +71,7 @@ the main_model name, suffixed by *_demo.xml* or *_data.xml*.
...
 
-For *static files*, since the resources can be used in different contexts (frontend, backend, both), they will be included in only one bundle. So, CSS/Less, JavaScript and XML files should be suffixed with the name of the bundle type. i.e.: *im_chat_common.css*, *im_chat_common.js* for 'assets_common' bundle, and *im_chat_backend.css*, *im_chat_backend.js* for 'assets_backend' bundle.
-If the module owns only one file, the convention will be *<module_name>.ext* (i.e.: *project.js*).
+For *static files*, the Javascript files start to follow the same logic as the python models. Each component should be in its own file, with a meaningfull name. For instance, the activity widgets are located in *activity.js* of mail module. Subdirectories can also be created to structure the 'package' (see web module for more details). The same logic should be apply for the templates of JS widgets (static XML files) and for their styles (scss files).
```
Comment (pedrobaeza) [inline doc/reference/guidelines.rst]: s/meaningfull/meaningful

Comment (pedrobaeza) [reply doc/reference/guidelines.rst]: Also s/be apply/be applied

File: doc/reference/guidelines.rst [hunk 3]
```diff
@@ -49,108 +50,186 @@ Other optional directories compose the module.
...
-Don't link data (image, libraries) outside Odoo: do not use an
-URL to an image but copy it in our codebase instead.
+.. code-block:: text
 
-Regarding *data*, split them by purpose: data or demo. The filename will be
-the *main_model* name, suffixed by *_data.xml* or *_demo.xml*.
+    addons/plant_nursery/
+    |-- security/
+    |   |-- ir.model.access.csv
+    |   |-- plant_nusery_groups.xml
```
Comment (fmdl) [inline doc/reference/guidelines.rst]: I think that, you may put `plant_nusery_groups.xml` before the `ir.model.access.csv`, if there are some new groups in `plant_nusery_groups.xml` (in the manifest file).

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

File: doc/reference/guidelines.rst [hunk 4]
```diff
@@ -49,108 +50,186 @@ Other optional directories compose the module.
...
+    |   |-- plant_nusery_security.xml
```
Comment (mreficent) [inline doc/reference/guidelines.rst]: typo: should be plant_nu**r**sery_groups.xml and plant_nu**r**sery_security.xml

**Pull request closed without merging**